code/src/sixsq/nuvla/server/resources/configuration_nuvla.clj

```diff
@@ -55,22 +74,24 @@ default values.
                          :template      {:href tpl-instance-url}}]
     (std-crud/add-if-absent config-instance-url p/resource-type create-template)
     (try
-      (when-let [stripe-api-key (or (-> config-instance-url
-                                        crud/retrieve-by-id-as-admin
-                                        :stripe-api-key)
-                                    (env/env :stripe-api-key))]
-        (let [pricing-instance (some-> "sixsq.nuvla.pricing.stripe.stripe" dyn/load-ns pricing-impl/set-impl!)]
-          (if (some? pricing-instance)
-            (do
-              (pricing-impl/set-impl! pricing-instance)
-              (pricing-impl/set-api-key! stripe-api-key)
-              (alter-var-root #'*stripe-api-key* (constantly stripe-api-key))
-              (when-let [stripe-client-id (or (-> config-instance-url
-                                                  crud/retrieve-by-id-as-admin
-                                                  :stripe-client-id)
-                                              (env/env :stripe-client-id))]
-                (alter-var-root #'*stripe-client-id* (constantly stripe-client-id))))
-            (log/error "Stripe-api-key configured but no princing implementation found!"))))
+      (let [nuvla-config (-> config-instance-url
+                             crud/retrieve-by-id-as-admin)]
+        (when-let [stripe-api-key (or (:stripe-api-key nuvla-config)
+                                      (env/env :stripe-api-key))]
+          (let [pricing-instance (some-> "sixsq.nuvla.pricing.stripe.stripe" dyn/load-ns pricing-impl/set-impl!)]
+            (if (some? pricing-instance)
+              (do
+                (pricing-impl/set-impl! pricing-instance)
+                (pricing-impl/set-api-key! stripe-api-key)
+                (alter-var-root #'*stripe-api-key* (constantly stripe-api-key))
+                (when-let [stripe-client-id (or (-> config-instance-url
+                                                    crud/retrieve-by-id-as-admin
+                                                    :stripe-client-id)
+                                                (env/env :stripe-client-id))]
+                  (alter-var-root #'*stripe-client-id* (constantly stripe-client-id))))
+              (log/error "Stripe-api-key configured but no princing implementation found!"))))
+        (when-let [authorized-redirect-urls (:authorized-redirect-urls nuvla-config)]
```

Comment:
I suggest placing this outside of the wrapping try/catch for loading Strype config, and creating its own try/catch if needed.